### ISSUE TITLE
How to ignore contents from subfolders ?

### ISSUE THREAD

**Issue by ngirard** [watchexec/watchexec#174]
I need to watch for any changes in html files from such `<d>` folder, while ignoring everything belonging `<d>`'s subfolders.

I tried invoking
```sh
watchexec -p -w <d> -f '*.html' -i './**/'
```
but it doesn't work: for instance, a newly created `<d>/sub/file.html` is not ignored.
Is there a way of achieving this using the current feature set ? If so, could it be added to the *Usage examples* section of the README ? 

Thanks in advance !

**Comment by passcod** [comment]
You can't use `-f` for this but you can use `-e html` instead with a pattern like this:

```
watchexec -p -w dir -e html -i 'dir/*/**/*'
```

**Comment by ngirard** [comment]
@passcod, thanks for your feedback !

I have to say, I'm having a hard time with this one.
The [globset](https://docs.rs/globset/0.4.6/globset/#syntax) documentation doesn't provide guidance for this case, which I think is pretty common (I guess I might want to report it).
I did some tests and it appears that the ignore pattern cannot be simplified down to `./*/**/*`. The parent directory needs to appear at the beginning, like you said, so when invoking `watchexec` the parent directory must appear twice.
And the pattern `/*/**/*` isn't straightforward at all.

Could there be an opportunity here to increase watchexec's usability by adding an option dedicated to ignoring subfolders ?
In any case, this is really worth adding to the examples IMHO...

**Comment by ngirard** [comment]
> Could there be an opportunity here to increase watchexec's usability by adding an option dedicated to ignoring subfolders ?
> In any case, this is really worth adding to the examples IMHO...

To be clear, what I have in mind is a `--max-depth` option, just like [Ripgrep](https://github.com/BurntSushi/ripgrep) and [Fd](https://github.com/sharkdp/fd). That would be great.

**Comment by mattgreen** [comment]
> Could there be an opportunity here to increase watchexec's usability by adding an option dedicated to ignoring subfolders ?

Perhaps, though this is the first issue I've seen about this since the project started. Most users have a workflow whereby build artifacts are separated from non-build artifacts at a high level. CMake and other build systems require this.

Adding another CLI arg for this use case feels a bit like a kludge IMO.

> In any case, this is really worth adding to the examples IMHO

PRs gladly accepted!

**Comment by abitrolly** [comment]
[#243](https://github.com/watchexec/watchexec/pull/243/files) could be extended with more patterns.

**Comment by ngirard** [comment]
Wow ! Thank you so much @passcod !

**Comment by passcod** [comment]
Released in 2.1.0, finally!
